core/src/main/kotlin/com/malinskiy/marathon/report/summary/TestSummaryFormatter.kt

```diff
@@ -28,41 +34,77 @@ class TestSummaryFormatter {
 
-            stringBuilder.appendln("Test batches:")
+            stringBuilder.appendln("Test runs details:")
+
+            summary.batches.forEachIndexed { index, batch ->
+                val batchSummary = batch.toSummaryString(index, currentTestResult.test)
+                stringBuilder.append(batchSummary)
+                stringBuilder.appendln()
+            }
+        }
+
+        return stringBuilder.toString()
+    }
+
+    private fun Batch.toSummaryString(index: Int, currentTest: Test): String {
+        val stringBuilder = StringBuilder()
+        val deviceSerial = testResults.first().device.serialNumber
 
-            summary.batches.forEach { batch ->
-                val isCurrentBatch = batch.batchId == currentTestResult.batchId
-                val batchBulletSymbol = if (isCurrentBatch) ">" else "*"
-                val deviceSerial = batch.testResults.first().device.serialNumber
+        val testStatus = testResults
+            .firstOrNull { it.test == currentTest }
+            ?.status
+            ?.toString() ?: "UNKNOWN"
 
-                stringBuilder.appendln("\u00a0\u00a0$batchBulletSymbol ${batch.batchId.createShortBatchId()} (${batch.testResults.size} tests, $deviceSerial):")
+        val batchBullet = "${index + 1})"
 
-                batch.testResults.forEach { testResult ->
-                    val isCurrentTest = testResult.test == currentTestResult.test
-                    val testBulletSymbol = if (isCurrentTest) ">" else "*"
+        stringBuilder.appendln("=".repeat(80))
+        stringBuilder.appendln("\u00a0\u00a0$batchBullet $testStatus in batch #${batchId.createShortBatchId()} (${testResults.size} tests in batch, device: $deviceSerial)")
+        stringBuilder.appendln("=".repeat(80))
+        stringBuilder.appendln("Tests in the batch (executed in the same process):")
 
-                    stringBuilder.append("\u00a0\u00a0\u00a0\u00a0$testBulletSymbol ${testResult.test.toSimpleSafeTestName()}")
+        testResults.forEach { testResult ->
+            val isCurrentTest = testResult.test == currentTest
+            val testBulletSymbol = if (isCurrentTest) ">" else "*"
 
-                    val additionalInfo = listOfNotNull(
-                        testResult.status.toString(),
-                        testResult.createShortFailureDescription()
-                    )
-                    stringBuilder.append(" (${additionalInfo.joinToString()})")
-                    stringBuilder.appendln()
-                }
+            stringBuilder.append("\u00a0\u00a0$testBulletSymbol ${testResult.test.toSimpleSafeTestName()}")
 
-                stringBuilder.appendln()
-            }
+            val additionalInfo = listOfNotNull(
+                testResult.status.toString(),
+                testResult.createShortFailureDescription()
+            )
+            stringBuilder.append(" (${additionalInfo.joinToString()})")
+            stringBuilder.append("\u00a0\u00a0")
```

Comment:
Why do you need to add two spaces at the end of the line?

Reply:
Fixed, thanks